recipes/generate.py

```diff
@@ -0,0 +1,176 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+import sys
+import time
+from typing import Any, Dict
+
+import torch
+from omegaconf import DictConfig
+
+from torch import nn
+
+from torchtune import config, utils
+
+logger = utils.get_logger("DEBUG")
+
+
+class InferenceRecipe:
```

Comment:
Can probably add documentation here around what's supported and what's not (e.g. explicitly call out we don't have speculative decoding, we support temperature and top-k sampling, etc.)